File: app/src/main/java/br/com/yourapp/pokeapi/models/Abilities.java
package br.com.yourapp.pokeapi.models;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.List;

public class Abilities implements Serializable {
    @SerializedName("ability")
    public Ability ability;

    @SerializedName("is_hidden")
    public boolean is_hidden;

    @SerializedName("slot")
    public int slot;

    public class Ability implements Serializable {
        @SerializedName("name")
        public String name;

        @SerializedName("url")
        public String url;
    }
}

File: app/src/main/java/br/com/yourapp/pokeapi/models/Abilities.kt
package br.com.yourapp.pokeapi.models

import com.google.gson.annotations.SerializedName
import java.io.Serializable

class Abilities : Serializable {
    @SerializedName("ability")
    var ability: Ability? = null

    @SerializedName("is_hidden")
    var is_hidden = false

    @SerializedName("slot")
    var slot = 0

    inner class Ability : Serializable {
        @SerializedName("name")
        var name: String? = null

        @SerializedName("url")
        var url: String? = null
    }
}